File: src/integration-tests/expecteds/py/circular-reference.py
from typing import TypedDict
from typing import Optional
from typing import NewType
from typing import Union

StringDoaGddGA = NewType("StringDoaGddGA", str)

OneOfMoebiusSchemaStringDoaGddGANK2MA6NR = NewType("OneOfMoebiusSchemaStringDoaGddGANK2MA6NR", Union[MoebiusSchema, StringDoaGddGA])

class MoebiusSchema(TypedDict):
    MoebiusProperty: Optional[MoebiusSchema]
    deeperMobiusProperty: Optional[OneOfMoebiusSchemaStringDoaGddGANK2MA6NR]

File: src/integration-tests/expecteds/ts/circular-reference.ts
export type StringDoaGddGA = string;
export type OneOfMoebiusSchemaStringDoaGddGANK2MA6NR = MoebiusSchema | StringDoaGddGA;
export interface MoebiusSchema {
  MoebiusProperty?: MoebiusSchema;
  deeperMobiusProperty?: OneOfMoebiusSchemaStringDoaGddGANK2MA6NR;
  [k: string]: any;
}
